tokio-util/src/net/mod.rs

```diff
@@ -40,7 +40,11 @@ impl Listener for tokio::net::TcpListener {
     }
 
     fn local_addr(&self) -> Result<Self::Addr> {
-        self.local_addr().map(Into::into)
+        #[cfg(not(target_os = "wasi"))]
```

Comment:
I don't think is necessary, although it's not supported we can leave this up to Mio/std, which doesn't support it, but we this way we don't need a `cfg` here).

Reply:
Done